### DISCUSSION TITLE
How to have more than one color for ascii logo

### DISCUSSION THREAD

**Discussion by tdtooke** [fastfetch-cli/fastfetch#444] [Q&A] [answered]
I want to have a theme that is something like my old neofetch theme.  I figured out how to set the logo color.  I see we have 9 indexes but I don't know how to paint the top half with one index and the bottom half with another.  I used --logo-color-1 31 and --logo-color-2 32 but it just sets the whole thing as 31(purple).

What I want:
![proper colors](https://user-images.githubusercontent.com/72276499/230796414-467b2d10-13f1-45fc-a632-fa873a916ef1.png)
 
What I currently have:

![colors](https://user-images.githubusercontent.com/72276499/230796420-9ad3c3ff-7b60-4103-a284-284914a23510.png)

**Comment by tdtooke** [comment]
Just to see if I'm right I decided to not be lazy and actually looked under the hood so to speak.  It looks like currently the problem is that there is only one thing for FF_LOGO_COLORS for the Arch Icon therefore I literally cannot do that.  If I another color to FF_LOGO_COLORS in Arch in builtin.c then redrew the icon so it used those 2 indexes where I want them the way some of the multicolor logos look it should work....I think?  Just want to know if I correctly understand the situation.  Of course this would obligate me to forever build this from source and apply a patch I made unless there is actually an interest in a multicolored Arch ascii logo here, there might not be.

Edit:  Just looked at neofetch and the Arch logo in it does have 2 color indexes.  So if I were to do this it would just be making the display the same as it is for neofetch in this case.  Somebody let me know it this is something that would be liked and I'll fork it, do a pull request, yada yada.  On a side note this would be the first time in more than 20 years that I've actually contributed anything to open-source...

**Reply by LinusDierheimer** [reply, marked as answer]
Yes you can open a pull request for that. Give the arch linux a second color index, but make it default to the first color. This way the default look is not changed, but can be customized!

**Reply by tdtooke** [reply]
Thanks, when I get home and can make sure it works I'll do that.

Edit: Pull request is up.

**Comment by Gapoly** [comment]
> I want to have a theme that is something like my old neofetch theme. I figured out how to set the logo color. I see we have 9 indexes but I don't know how to paint the top half with one index and the bottom half with another. I used --logo-color-1 31 and --logo-color-2 32 but it just sets the whole thing as 31(purple).
> 
> What I want: ![proper colors](https://user-images.githubusercontent.com/72276499/230796414-467b2d10-13f1-45fc-a632-fa873a916ef1.png)
> 
> What I currently have:
> 
> ![colors](https://user-images.githubusercontent.com/72276499/230796420-9ad3c3ff-7b60-4103-a284-284914a23510.png)

Could you send me your config files. I'm trying to do some changes on mine but I suck at json *(and any langages)*, I have no idea how to changes things.

**Reply by tdtooke** [reply]
I can do that when I get home.  The most important thing to remember is to have the link to the schema at the top of your json file so if you're using something like Code you get autocomplete options.  I don't think I could do this at all without that.  Also on mine you probably want to remove the 'vulkan' bit.  There was a time when it stopped detecting my discrete GPU without it.  That might no longer be the case but at any rate if you don't have 2 GPUs that will only slow it down for no reason.

**Comment by tdtooke** [comment]
Ok, here is the config I used while on Plasma.  My hyprland config is a work in progress.  For your needs if you're not Arch you may want to add more logo colors if your distro has more than 2 colors.
<code>{
  "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
  "logo": {
    "padding": {
      "top": 3,
      "left": 1
    },
    "color": {
      "1": "red",
      "2": "green"
    }
  },
  "display": {
    "color": {
      "title": "cyan"
    },
    "hideCursor": false,
    "separator": "   "
  },
  "modules": [
    "title",
    "separator",
    {
      "type": "host",
      "keyColor": "red",
      "key": " PC"
    },
    {
      "type": "board",
      "keyColor": "red",
      "key": "  ├"
    },
    {
      "type": "CPU",
      "keyColor": "red",
      "key": "  ├"
    },
    {
      "type": "gpu",
      "keyColor": "red",
      "detectionMethod": "vulkan",
      "key": "  ├﬙"
    },
    {
      "type": "display",
      "keyColor": "red",
      "key": "  ├"
    },
    {
      "type": "memory",
      "keyColor": "red",
      "key": "  ╰"
    },
    "break",
    {
      "type": "os",
      "keyColor": "green",
      "key": " OS"
    },
    {
      "type": "kernel",
      "keyColor": "green",
      "key": "  ├"
    },
    {
      "type": "lm",
      "keyColor": "green",
      "key": "  ├󰍂"
    },
    {
      "type": "shell",
      "keyColor": "green",
      "key": "  ├"
    },
    {
      "type": "terminal",
      "keyColor": "green",
      "key": "  ├"
    },
    {
      "type": "TerminalFont",
      "keyColor": "green",
      "key": "  ├"
    },
    {
      "type": "Packages",
      "keyColor": "green",
      "key": "  ╰"
    },
    "break",
    {
      "type": "DE",
      "keyColor": "yellow",
      "key": " DE"
    },
    {
      "type": "WM",
      "keyColor": "yellow",
      "key": "  ├"
    },
    {
      "type": "WMTheme",
      "keyColor": "yellow",
      "key": "  ├"
    },
    {
      "type": "Theme",
      "keyColor": "yellow",
      "key": "  ├"
    },
    {
      "type": "Icons",
      "keyColor": "yellow",
      "key": "  ├"
    },
    {
      "type": "Font",
      "keyColor": "yellow",
      "key": "  ├"
    },
    {
      "type": "Cursor",
      "keyColor": "yellow",
      "key": "  ╰󰇀"
    },
    "break",
    {
      "type": "colors",
      "paddingLeft": 3,
      "symbol": "diamond"
    }
  ]
}</code>
